e2e/test/iothub/command/CommandE2ETests.cs

```diff
@@ -0,0 +1,323 @@
+                    try
+                    {
+                        var valueToTest = request.GetData<ServiceCommandRequestObject>();
+                        Assert.AreEqual(methodName, $"{request.ComponentName}*{request.CommandName}", $"The expected method name should be {methodName} but was {request.ComponentName}*{request.CommandName}");
+                        Assert.AreEqual(ServiceRequestJson, request.DataAsJson, $"The expected respose payload should be {ServiceRequestJson} but was {request.DataAsJson}");
+                        Assert.AreEqual(ServiceCommandRequestAssertion.a, valueToTest.a, $"The expected respose object did not decode properly. Value a should be {ServiceCommandRequestAssertion.a} but was {valueToTest?.a ?? int.MinValue}");
+                        methodCallReceived.SetResult(true);
+                    }
+                    catch (Exception ex)
+                    {
+                        methodCallReceived.SetException(ex);
+                    }
+
+                    return Task.FromResult(new CommandResponse(new DeviceCommandResponse(), 200));
+                },
+                null).ConfigureAwait(false);
+
+            // Return the task that tells us we have received the callback.
+            return methodCallReceived.Task;
+        }
+
+        private async Task SendCommandAndUnsubscribeAsync(Client.TransportType transport, Func<DeviceClient, string, MsTestLogger, Task<Task>> subscribeAndUnsubscribeMethod, bool withComponent = false, TimeSpan responseTimeout = default, ServiceClientTransportSettings serviceClientTransportSettings = default)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            await subscribeAndUnsubscribeMethod(deviceClient, CommandName, Logger).ConfigureAwait(false);
+
+            await ServiceSendCommandAndVerifyResponseAsync(testDevice.Id, CommandName, Logger, responseTimeout: responseTimeout, serviceClientTransportSettings: serviceClientTransportSettings).ConfigureAwait(false);
+
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+        }
+
+        private async Task SendCommandAndRespondAsync(Client.TransportType transport, Func<DeviceClient, string, MsTestLogger, Task<Task>> setDeviceReceiveMethod, bool withComponent = false, TimeSpan responseTimeout = default, ServiceClientTransportSettings serviceClientTransportSettings = default)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            string commandNameString = CommandName;
+            if (withComponent)
+            {
+                commandNameString = $"{ComponentName}*{CommandName}";
```

Comment:
How about we use the pnp specific service client for the service operation; then we wouldn't need to do this string manipulation for command name: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/service/src/DigitalTwin/DigitalTwinClient.cs#L235